File: assessment-3/app/dto/movie.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class MovieDTO:
    """Data Transfer Object for Movie."""
    movie_name: str
    movie_link: str
    fshare_link: str
    original_title: str
    genre: str
    year: Optional[int]
    runtime: Optional[str]
    rating: float
    overview: str
    poster_url: str
    backdrop_url: str
    tmdb_id: str
    category_id: Optional[int] = None

File: assessment-4/movie-fe/src/types/movie.ts
export interface Movie {
  movie_name: string;
  movie_link: string;
  fshare_link: string;
  original_title: string;
  genre: string;
  year: number | null;
  runtime: string | null;
  rating: number;
  overview: string;
  poster_url: string;
  backdrop_url: string;
  tmdb_id: string;
  category_id: number | null;
}

export interface Category {
  id: number;
  name: string;
  subcategories?: Category[];
}
